src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+        effectPosition.z = this->actor.posRot.pos.z + (new_var->z * this->actor.scale.z * 40.0f);
+        fragmentVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 6.0f;
+        fragmentVelocity.y = Math_Rand_ZeroOne() * 10.0f;
+        fragmentVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 6.0f;
+        index = (s32)(Math_Rand_ZeroOne() * 111.1f) % 7;
+        EffectSsKakera_Spawn(globalCtx, &effectPosition, &fragmentVelocity, &effectPosition, -100, 64, 40, 3, 0,
+                             D_80A9C26C[index], 0, 0, 80, KAKERA_COLOR_NONE, OBJECT_GAMEPLAY_KEEP, D_040356A0);
+    }
+}
+
+// Spawn Bugs
+void func_80A9B574(EnKusa* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        Actor* bug =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, Math_Rand_ZeroOne() * 0xFFFF, 0, 1);
+
+        if (bug == NULL) {
+            break;
+        }
+    }
+}
+
+// EnKusa_InitCollider
+void func_80A9B630(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80A9C208);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+}
+
+void EnKusa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, D_80A9C27C);
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    // Init for the ColliderCylinder
+    func_80A9B630(thisx, globalCtx);
+
+    func_80061ED4(&this->actor.colChkInfo, NULL, &D_80A9C234);
+    if (this->actor.shape.rot.y == 0) {
+        // Random value I think?
+        s32 rand = (s32)Math_Rand_ZeroFloat(65536.0f);
+        this->actor.posRot.rot.y = (s16)rand;
+        this->actor.initPosRot.rot.y = (s16)rand;
+        this->actor.shape.rot.y = (s16)rand;
+    }
+
+    if (func_80A9AFAC(this, globalCtx, 0.0f) == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->kusaTexObjIndex = Object_GetIndex(&globalCtx->objectCtx, D_80A9C200[thisx->params & 3]);
+
+    if (this->kusaTexObjIndex < 0) {
+        // Prints -> Bank danger!
+        osSyncPrintf("Error : バンク危険！ (arg_data 0x%04x)(%s %d)\n", thisx->params, "../z_en_kusa.c", 0x231);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    // Maybe this is SetupWait?
+    func_80A9B7EC(this);
+}
+
+void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+
+    Collider_DestroyCylinder(globalCtx, &THIS->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B07C.s")
+// SetupWait?
+void func_80A9B7EC(EnKusa* this) {
+    func_80A9AFA0(this, func_80A9B810);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B140.s")
+void func_80A9B810(EnKusa* this, GlobalContext* globalCtx) {
```

Comment:
This function is a common construction for actors that require additional objects. Checkout the BgMori actors for examples if you want to name it.